### ISSUE TITLE
@batchcode fails with tutorial_examples.example_batch_code.py

### ISSUE THREAD

**Issue by ergodicbreak** [evennia/evennia#909]
Running a clean checkout of current Evennia, I start up the game, run `@batchcode example_batch_code` and get:

```
[MudInfo] [MudInfo, 2015-12-31(03:51)]: Evennia connected

You become Evennia.
Limbo(#2)
Welcome to your new Evennia-based game! Visit http://www.evennia.com if you need
help, want to contribute, report issues or just join the community.
As Player #1 you can create a demo/tutorial area with @batchcommand tutorial_world.build.
Running Batch-code processor - Automatic mode for example_batch_code ...
01/03: #CODE : 

from evennia import create_object, search_obj[...]
02/03: #CODE (create red button): 

red_button = create_object[...]
>>> 
>>> 03: except RuntimeError: pass
>>> 04: finally: del settings_module
>>> 05: #CODE (create red button): 
>>> 06: 
>>> 07: red_button = create_object(red_button.RedButton, key="Red button",
>>> 08:                                           location=limbo, aliases=["button"])
>>> 09: caller.msg("A %s was created." % red_button.key)
Traceback (most recent call last):
>>> 
>>>   File "<string>", line 7, in <module>
>>> 
>>> NameError: name 'create_object' is not defined
>>>
```

I have a fix for this which I'll submit as a PR, but it involves a substantial change to the batchcode parser so I want to make sure someone else can reproduce and it's not just me.

**Comment by Drevarr** [comment]
Installation with the tutorial world added.  Output from the following ```@batchcode example_batch_code``` and ```@batchcode/debug example_batch_code```.

```
Running Batch-code processor - Automatic mode for example_batch_code ...
01/03: #CODE :

from evennia import create_object, search_obj[...]
02/03: #CODE (create red button):

red_button = create_object[...]
>>>
>>> 03: except RuntimeError: pass
>>> 04: finally: del settings_module
>>> 05: #CODE (create red button):
>>> 06:
>>> 07: red_button = create_object(red_button.RedButton, key="Red button",
>>> 08:                                           location=limbo, aliases=["button"])
>>> 09: caller.msg("A %s was created." % red_button.key)
Traceback (most recent call last):
>>>
>>>   File "<string>", line 7, in <module>
>>>
>>> NameError: name 'create_object' is not defined
>>>
```

and

```
Running Batch-code processor - Automatic mode for example_batch_code ...
01/03: #CODE :

from evennia import create_object, search_obj[...]
02/03: #CODE (create red button):

red_button = create_object[...]
>>> Traceback (most recent call last):
>>>
>>>   File "c:\users\john\mudev\evennia\evennia\utils\batchprocessors.py", line 421, in code_exec
>>>     exec(code, environdict)
>>>
>>>   File "<string>", line 11
>>>
>>>     .delete()
>>>
>>>     ^
>>>
>>> SyntaxError: invalid syntax
>>>
```

**Comment by Griatch** [comment]
Hm, is this still an issue? I just try `@batchcode example_batch_code` and it seems to work fine ...

**Comment by ergodicbreak** [comment]
@Griatch weird -- I haven't looked at this since I opened the issue. I don't suppose it could be a platform issue? I'm on Windows. @Drevarr what did you test on?

**Comment by CloudKeeper1** [comment]
I've come across the same issue. My initial code was just testing Batchcode:
```
#HEADING
from evennia import create_object, search_object
from typeclasses import rooms

#CODE

testroom = create_object(rooms.Room, key="testroom", aliases=["room"])
testroom.db.desc = "Testing"
```
Which returned:
```
from evennia import create_object, search_obj[...]
02/02: #CODE : 

testroom = create_object(rooms.Room, key="tes[...]
>>> 
>>> 03: except RuntimeError: pass
>>> 04: finally: del settings_module
>>> 05: #CODE : 
>>> 06: 
>>> 07: testroom = create_object(rooms.Room, key="testroom", aliases=["room"])
>>> 08: testroom.db.desc = "Testing"Traceback (most recent call last):
>>> 
>>>   File "<string>", line 7, in <module>
>>> 
>>> NameError: name 'create_object' is not defined
>>>
```
Next step was testing the example_batch_code:
```
from evennia import create_object, search_obj[...]
02/03: #CODE (create red button): 

red_button = create_object[...]
>>> 
>>> 03: except RuntimeError: pass
>>> 04: finally: del settings_module
>>> 05: #CODE (create red button): 
>>> 06: 
>>> 07: red_button = create_object(red_button.RedButton, key="Red button",
>>> 08:                                           location=limbo, aliases=["button"])
>>> 09: caller.msg("A %s was created." % red_button.key)
Traceback (most recent call last):
>>> 
>>>   File "<string>", line 7, in <module>
>>> 
>>> NameError: name 'create_object' is not defined
>>>
```
I'm on Windows 7 using python 2.7.11, evennia 0.5.0 (rev 5d6d13b)
The obvious observation is that the Header isn't being applied to the Code blocks..

**Comment by Griatch** [comment]
I cannot reproduce this at this time, and it sounds strange that it would be a Windows-unique issue. @ergodicbreak, if you have a fix that solves this, that in itself might give a hint as to what is the problem.

**Comment by Drevarr** [comment]
Still an issue on Windows 10. Just cloned latest master and setup this evening.

```
[MudInfo] [MudInfo, 2016-01-30(03:11)]: drevarr connected

You become drevarr.
Limbo(#2)
Welcome to your new Evennia-based game! Visit http://www.evennia.com if you need
help, want to contribute, report issues or just join the community.
As Player #1 you can create a demo/tutorial area with @batchcommand tutorial_world.build.
Running Batch-code processor - Automatic mode for example_batch_code ...
01/03: #CODE :

from evennia import create_object, search_obj[...]
02/03: #CODE (create red button):

red_button = create_object[...]
>>>
>>> 03: except RuntimeError: pass
>>> 04: finally: del settings_module
>>> 05: #CODE (create red button):
>>> 06:
>>> 07: red_button = create_object(red_button.RedButton, key="Red button",
>>> 08:                                           location=limbo, aliases=["button"])
>>> 09: caller.msg("A %s was created." % red_button.key)
Traceback (most recent call last):
>>>
>>>   File "<string>", line 7, in <module>
>>>
>>> NameError: name 'create_object' is not defined
>>>
```

**Comment by CloudKeeper1** [comment]
Just as an update. This is still an issue with the current version of Evennia.

**Comment by benabhi** [comment]
Running Batch-code processor - Automatic mode for generate_world ...
01/01: from typeclasses import rooms, exits\n\nidentifier = "AB1[...]
>>> 
>>> 07: 
>>> 08: from typeclasses import rooms, exits
>>> 09: 
>>> 10: identifier = "AB105"
>>> 11: a = create_object(rooms.Room, key=identifier)
>>> 12: 
>>> 13: 
>>> 14: #letters = "ABCDEFGHIJKLMNnOPQRSTUVWXYZ"
>>> 15: #matrix = [[0] * 5 for i in range(5)]Traceback (most recent call last):
>>> 
>>>   File "<string>", line 11, in <module>
>>> 
>>> NameError: name 'create_object' is not defined


Same issue here

**Comment by Griatch** [comment]
The batchcode parser was completely reworked since the original issue, interesting if the error still remains across to the new version. Reopening.

**Comment by benabhi** [comment]
I'm stuck with that since yesterday. I really need this feature becose i try to make a bundle of rooms with loops.

This simple sinippet fail:

```python
from evennia import create_object, search_object
myroom = create_object("typeclasses.rooms.Room", key="myroom")
limbo  = search_object("Limbo")[0]
exit   = create_object("typeclasses.exits.Exit", key="norte", location=limbo, destination=myroom)
```

with this error:

<pre>
@batchcode generate_world
Running Batch-code processor - Automatic mode for generate_world ...
01/01: myroom = create_object("typeclasses.rooms.Room", key="s[...]
>>> 
>>> 06: # batchcode code:
>>> 07: 
>>> 08: 
>>> 09: 
>>> 10: myroom = create_object("typeclasses.rooms.Room", key="sala")
>>> 11: limbo  = search_object("Limbo")[0]
>>> 12: exit   = create_object("typeclasses.exits.Exit", key="norte", location=limbo, destination=myroom)
>>> 13: Traceback (most recent call last):
>>> 
>>>   File "<string>", line 10, in <module>
>>> 
>>> NameError: name 'create_object' is not defined
>>>
</pre>

Some things i tried:

* Reinstall my entire development environment
* Import objects like rooms.Room for fist paramter (create_object)
* Import evennia.utils import create

My specs:
Evennia version: 0.60
System: Mint 18 / codename: sarah
Python: 2.7.12

It's the only thing that does not work for me.

**Comment by Griatch** [comment]
@benabhi That's a curious issue, not necessarily related to the original one. Haven't had time to look into it yet though.

**Comment by vlegoff** [comment]
I can run `@batchcode tutorial_examples.example_batch_code` from a fresh install with no error.  I get the red button, table and chair.  I'm not sure this issue, if it's still active, is really OS-dependent:

- System: Windows 10 (64-bit)
- Python version: 2.7.13
- Evennia version: 0.6 (last from master)

**Comment by Griatch** [comment]
This should be fixed with the merger of `devel` and Evennia 0.7, where the batchcode parser was reworked. Please comment here if the problem persists.
